tests/spec/s-object.js

```diff
@@ -356,4 +361,35 @@ describe('Object', function () {
             expect(obj instanceof Object).toBe(false);
         });
     });
+
+    describe('#toString', function () {
+        it('basic', function () {
+            expect(toStr.call()).toBe('[object Undefined]');
+            /*eslint no-useless-call: 0*/
+            expect(toStr.call(undefined)).toBe('[object Undefined]');
+            expect(toStr.call(null)).toBe('[object Null]');
+            expect(toStr.call(1)).toBe('[object Number]');
+            expect(toStr.call(true)).toBe('[object Boolean]');
+            expect(toStr.call('x')).toBe('[object String]');
+            expect(toStr.call([1, 2, 3])).toBe('[object Array]');
+            expect(toStr.call(returnArgs())).toBe('[object Arguments]');
+            expect(toStr.call({})).toBe('[object Object]');
+            expect(toStr.call(noop)).toBe('[object Function]');
+            expect(toStr.call(new RegExp('c'))).toBe('[object RegExp]');
+            expect(toStr.call(new Date())).toBe('[object Date]');
+            expect(toStr.call(new Error('x'))).toBe('[object Error]');
+        });
+
+        it('prototypes', function () {
```

Comment:
this test will actually cause problems in future implementations, since in ES6, many of these prototype objects stop being instances of themselves, and instead would return `'[object Object]'`.

Please split these up such that array/boolean/object/function are always tested, but that the rest are skipped with an `xit`, much like https://github.com/es-shims/es5-shim/blob/master/tests/spec/s-object.js#L89-L90